### ISSUE TITLE
Show important parameters based on non-linear process and method selected

### ISSUE THREAD

**Issue by klauer** [pcdshub/las-dispersion-scan#22]
## Expected Behavior
Show only the used/important parameters for each retrieval/non-linear process.

## Current Behavior
Shows all parameters due to author's lack of knowledge about what's important

## Context
Per https://github.com/pcdshub/las-dispersion-scan/issues/18

> Two parameters, "non-linear process" (existing) and "method" (not existing) are important for the retrieval. The non-linear process can be fed directly into the retrieval algorithm. However, the method is used for preparing the data, and depending on the method, other parameters are needed too. For example, in d-scan, we need the material, wedge angle, and motor geometry. It would probably be easiest for us to meet to discuss this, but a specific example for the OPCPA is in the thread of this message."

**Comment by brittonm** [comment]
Correct! I am currently working on a phase retrieval algorithm to replace Pypret that will have the method parameters much better defined, so it might make sense to wait for a version of my algorithm that will be ready in a month or two. As it stands, we do not fully trust Pypret, so it does not make sense to put too much time into integrating it further.

We could start by creating drop down options for "non-linear process" (e.g. SHG, THG, SD, XPW) and "method" (e.g. FROG, material dispersion scan, grating dispersion scan). Then, the chosen method would need more parameters to be defined. In the case of material dispersion scan, three additional parameters need to be defined: the material, wedge angle, and motor geometry.
